src/read/elf/note.rs

```diff
@@ -122,12 +122,11 @@ impl<'data, Elf: FileHeader> Note<'data, Elf> {
     /// The length of this field (including any null terminator) is given by
     /// `n_namesz`.
     pub fn name(&self) -> &'data [u8] {
-        if let Some((last, name)) = self.name.split_last() {
-            if *last == 0 {
-                return name;
-            }
+        let mut name = self.name;
+        while let [rest @ .., 0] = name {
```

Comment:
Technically this is a breaking change, and it at least needs a documentation update to indicate that it excludes all trailing null bytes.

An alternative would be to add a `name_bytes` function that doesn't strip any terminators, and define `ELF_NOTE_GO` to be `b"Go\0\0"`. We should probably add `name_bytes` either way.

Reply:
Updated documentation and added `name_bytes`, but have left the definition of `ELF_NOTE_GO` unchanged for consistency with the other notes. Let me know if you prefer that I change those as well, but that'd be a breaking change so I suspect you'd prefer they remain as they are.